src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1746,6 +1746,10 @@ private function registerMessengerConfiguration(array $config, ContainerBuilder
 
             // make sure senderAliases contains all senders
             foreach ($messageConfiguration['senders'] as $sender) {
+                if (!isset($senderAliases[$sender])) {
```

Comment:
that's the same check as just below, strange.

Reply:
I think that second check needed for routing message using transport service id, for example instead of `amqp` route to `messenger.transport.amqp`. Should we support this case? [Oficial guide](https://symfony.com/doc/current/messenger.html#routing-messages-to-a-transport) says nothing about possibility to routing messages using service id.

Reply:
But as you throw, the second condition will never be true, right?